src/RA_Leaderboard.cpp

```diff
@@ -258,48 +235,30 @@ void RA_Leaderboard::ParseLBData( char* pChar )
 		if( std::string( "STA:" ).compare( 0, 4, pChar, 0, 4 ) == 0 )
 		{
 			pChar += 4;
-
-			//	Parse Start condition
-			do 
-			{
-				while( ( *pChar ) == ' ' || ( *pChar ) == '_' || ( *pChar ) == '|' )
-					pChar++; // Skip any chars up til this point :S
-				
-				Condition nNewCond;
-				nNewCond.ParseFromString( pChar );
-				m_startCond.Add( nNewCond );
-			}
-			while( *pChar == '_' );
+			m_startCond.ParseFromString(pChar);
 		}
 		else if( std::string( "CAN:" ).compare( 0, 4, pChar, 0, 4 ) == 0 )
 		{
 			pChar += 4;
-			//	Parse Cancel condition
-			do 
+			m_cancelCond.ParseFromString(pChar);
+
+			// temporary backwards compatibility support: all conditions in CANCEL should be OR'd:
+			if (m_cancelCond.GroupCount() == 1 && m_cancelCond.GetGroup(0).Count() > 1)
 			{
-				while( ( *pChar ) == ' ' || ( *pChar ) == '_' || ( *pChar ) == '|' )
-					pChar++; // Skip any chars up til this point :S
-				
-				Condition nNewCond;
-				nNewCond.ParseFromString( pChar );
-				m_cancelCond.Add( nNewCond );
+				for (size_t i = 0; i < m_cancelCond.GetGroup(0).Count(); ++i)
+				{
+					m_cancelCond.AddGroup();
+					m_cancelCond.GetGroup(i + 1).Add(m_cancelCond.GetGroup(0).GetAt(i));
+				}
+
+				m_cancelCond.GetGroup(0).Clear();
 			}
-			while( *pChar == '_' );
+			// end backwards compatibility conversion
```

Comment:
This block (lines 245-256) should be removed once existing leaderboard cancel conditions are converted.